pkg/controllers/capr/managesystemagent/managesystemagent.go

```diff
@@ -122,9 +124,13 @@ func (h *handler) OnChange(cluster *rancherv1.Cluster, status rancherv1.ClusterS
 		return nil, status, err
 	}
 
+	if cluster.Status.FleetWorkspaceName == "" {
```

Comment:
good catch. In this case, I should be returning a `generic.ErrSkip` as this field can temporarily be unset on upgrade. returning a `nil` err would cause the bundle to be removed erroneously.